Installation.md

```diff
@@ -1,76 +1,139 @@
+TODO: the configuration file can also be in the Flask instance directory.
 
-* Make sure you have a Postgres (Version 9+) database. See `data/Readme.md` for instructions on this.
-* Tell `flexmeasures` about it. Either you are using the default for the environment you're in (see `flexmeasures/utils/config_defaults`),
-   or you can configure your own connection string: In `flexmeasures/<development|testing|staging|production>_conf.py`,
-  set the variable `SQLALCHEMY_DATABASE_URI = 'postgresql://<user>:<password>@<host-address>[:<port>]/<db>'`
-* Run `flask db upgrade` to create the Postgres DB structure.
 
-## Preparing the job queue database
+### Configure environment
 
-To let FlexMeasures queue forecasting and scheduling jobs, install a Redis server and configure access to it within FlexMeasures' config file (see above). You can find the default settings in `flexmeasures/utils/config_defaults.py`.
+Set an environment variable to indicate in which environment you are operating (one out of development|testing|staging|production), e.g.:
 
-TODO: more detail
+   `echo "FLASK_ENV=development" >> .env`
 
-## Install an LP solver
+or:
 
-For planning balancing actions, the flexmeasures platform uses a linear program solver. Currently that is the Cbc solver. See the `FLEXMEASURES_LP_SOLVER` config setting if you want to change to a different solver.
+   `export FLASK_ENV=development`
 
-Installing Cbc can be done on Unix via:
+(on Windows, use `set` instead of `export`)
 
-    apt-get install coinor-cbc
+Note: The default is `production`, which will not work well on localhost due to SSL issues. 
 
-(also available in different popular package managers).
 
-We provide a script for installing from source (without requiring `sudo` rights) in [the CI Readme](ci/Readme.md).
+### Make a secret key for sessions and password salts
+
+Set a secret key which is used to sign user sessions and re-salt their passwords. The simplest way is again with an environment variable, like this:
+
+   `export "SECRET_KEY=something-secret"`
+
+(on Windows, use `set` instead of `export`)
+
+Actually, we recommend you add this setting to your config file (see above). Here is a Pythonic way to generate good random string for this:
+
+    `python -c "import secrets; print(secrets.token_urlsafe())"`
+
+
+### Add a user
 
-More information (e.g. for installing on Windows) on [the website](https://projects.coin-or.org/Cbc).
+FlexMeasures is a web-based platform, so we need a user account:
 
+`flexmeasures new-user --username <your-username> --email <your-email-address>`
 
-## Run
+(this will ask you to set a password for the user)
 
-Now, to start the web application, you can run:
 
-    python flexmeasures/run-local.py
+### Add structure
 
-But in a production context, you shouldn't run a script - hand the `app` object to a WSGI process, as your platform of choice describes.
+Populate the database with some standard energy asset types:
+
+   `flexmeasures db_populate --structure`
```

Comment:
Another failed step for me. Full traceback this time:

```
Warning: Without complete mail settings, FlexMeasures will not be able to send mails to users, e.g. for password resets!
You might consider setting MAIL_SERVER, MAIL_PORT, MAIL_USE_TLS, MAIL_USE_SSL, MAIL_USERNAME, MAIL_PASSWORD.
[FLEXMEASURES][2021-03-12 15:51:43,426] ERROR: [populate_structure] Encountered Problem: populate_structure() missing 1 required positional argument: 'db'
Traceback (most recent call last):
  File "/home/felix/my-fm-venv/bin/flexmeasures", line 10, in <module>
    sys.exit(flexmeasures_cli())
  File "/home/felix/miniconda3/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/felix/my-fm-venv/lib/python3.8/site-packages/flask/cli.py", line 586, in main
    return super(FlaskGroup, self).main(*args, **kwargs)
  File "/home/felix/miniconda3/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/felix/miniconda3/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/felix/miniconda3/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/felix/miniconda3/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/felix/miniconda3/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/felix/my-fm-venv/lib/python3.8/site-packages/flask/cli.py", line 426, in decorator
    return __ctx.invoke(f, *args, **kwargs)
  File "/home/felix/miniconda3/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/felix/my-fm-venv/lib/python3.8/site-packages/flexmeasures/data/scripts/cli_tasks/db_pop.py", line 121, in db_populate
    populate_structure(app.db)
  File "/home/felix/my-fm-venv/lib/python3.8/site-packages/flexmeasures/data/transactional.py", line 38, in wrap
    db_function(*args[1:], **kwargs)
TypeError: populate_structure() missing 1 required positional argument: 'db'
```

Reply:
I also expect `flexmeasures db current` to show the current database revision, but I just get the warning about missing mail settings and then nothing. For the expected behaviour, I tried in my venv on our main branch:
```
$ flask db current
22ce09690d23 (head)
```

Reply:
It works for me when I did run a successful migration, which I can now.